src/openrct2/peep/Staff.cpp

```diff
@@ -181,7 +181,7 @@ bool Staff::IsLocationInPatrol(const CoordsXY& loc) const
         return false;
 
     // Check if staff has patrol area
-    if (!(gStaffModes[StaffId] & 2))
+    if (gStaffModes[StaffId] != StaffMode::Patrol)
         return true;
 
     return IsPatrolAreaSet(loc);
```

Comment:
As weird as it sounds, I think this function is right. If a particular staff has no patrol areas set, all areas that are asked of they will count as in their patrol, because they have no specific one defined.